src/org/labkey/test/params/FieldDefinition.java

```diff
@@ -195,147 +170,152 @@ public PropertyDescriptor setLookup(String schema, String query, String containe
 
     public FieldDefinition setValidators(List<FieldValidator<?>> validators)
     {
+        JSONArray propertyValidators = null;
+        if (validators != null)
+        {
+            propertyValidators = new JSONArray();
+            validators.stream().map(FieldValidator::toJSONObject).forEachOrdered(propertyValidators::add);
+        }
+        setFieldProperty("propertyValidators", propertyValidators);
         _validators = validators;
         return this;
     }
 
     public String getURL()
     {
-        return _url;
+        return (String) getFieldProperty("URL");
     }
 
     public FieldDefinition setURL(String url)
     {
-        _url = url;
+        setFieldProperty("URL", url);
         return this;
     }
 
     public String getImportAliases()
     {
-        return _importAliases;
+        return (String) getFieldProperty("importAliases");
     }
 
     public FieldDefinition setImportAliases(String importAliases)
     {
-        _importAliases = importAliases;
+        setFieldProperty("importAliases", importAliases);
         return this;
     }
 
     public Integer getScale()
     {
-        return _scale;
+        return (Integer) getFieldProperty("scale");
     }
 
-    @Override
-    public FieldDefinition setHidden(Boolean hidden)
+    public FieldDefinition setScale(Integer scale)
     {
-        super.setHidden(hidden);
+        setFieldProperty("scale", scale);
         return this;
     }
 
     public Boolean isPrimaryKey()
     {
-        return  _isPrimaryKey;
+        return (Boolean) getFieldProperty("isPrimaryKey");
```

Comment:
I'm sure this works, but just curious if it would be better/safer to do `Boolean.valueOf(getFieldProperty("isPrimaryKey"))`?

Reply:
We have full control of what goes into the property Map. Somebody would have to go out of their way to make this not work.